app/assets/javascripts/discourse/app/components/topic-admin-menu.gjs

```diff
@@ -0,0 +1,298 @@
+import Component from "@glimmer/component";
+import { fn } from "@ember/helper";
+import { action } from "@ember/object";
+import { inject as service } from "@ember/service";
+import { and, not, or } from "truth-helpers";
+import DButton from "discourse/components/d-button";
+import concatClass from "discourse/helpers/concat-class";
+import icon from "discourse-common/helpers/d-icon";
+import getURL from "discourse-common/lib/get-url";
+import DMenu from "float-kit/components/d-menu";
+
+export default class TopicAdminMenu extends Component {
+  @service adminTopicMenuButtons;
+  @service currentUser;
+
+  @action
+  onRegisterApi(api) {
+    this.dMenu = api;
+  }
+
+  @action
+  onButtonAction(buttonAction) {
+    this.args[buttonAction]?.();
+    this.dMenu.close();
+  }
+
+  @action
+  onExtraButtonAction(buttonAction) {
+    buttonAction?.();
+    this.dMenu.close();
+  }
+
+  get extraButtons() {
+    return this.adminTopicMenuButtons.callbacks
+      .map((callback) => {
+        return callback(this.args.topic);
+      })
+      .filter(Boolean);
+  }
+
+  get details() {
+    return this.args.topic.get("details");
+  }
+
+  get isPrivateMessage() {
+    return this.args.topic.get("isPrivateMessage");
+  }
+
+  get featured() {
+    return (
+      !!this.args.topic.get("pinned_at") || this.args.topic.get("isBanner")
+    );
+  }
+
+  get visible() {
+    return this.args.topic.get("visible");
+  }
+
+  get canDelete() {
+    return this.details.get("can_delete");
+  }
+
+  get canRecover() {
+    return this.details.get("can_recover");
+  }
+
+  get archived() {
+    return this.args.topic.get("archived");
+  }
+
+  get topicModerationHistoryUrl() {
+    return getURL(`/review?topic_id=${this.args.topic.id}&status=all`);
+  }
+
+  <template>
+    <span class="topic-admin-menu-button-container">
+      <span class="topic-admin-menu-button">
+        <DMenu
+          @onRegisterApi={{this.onRegisterApi}}
+          @triggerClass="toggle-admin-menu"
```

Comment:
We probably want: @modalForMobile={{true}}

You can also add it to the post menu one in the same PR please